File: IntentPassing/app/src/main/java/com/example/intentpassing/SecondActivity.java
package com.example.intentpassing;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_second);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.second), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        Intent fromMnAct = getIntent();                               //8th assign this activity Intent variable from the main activity
        String Name = fromMnAct.getStringExtra("Name");         //9th Store data by putting key to a variable
        int Age = fromMnAct.getIntExtra("Age",0);

        TextView details;                                             // making a variable where we show the data
        details = findViewById(R.id.details);                         // assign that variable to xml id
        details.setText("Name: "+Name+ "\nAge: " + Age);              // now set those data which we pull from the main activity
    }
}

File: MultipleActivities/app/src/main/java/com/example/multipleactivities/SecondActivity.kt
package com.example.multipleactivities

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity

class SecondActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_second)

        findViewById<Button>(R.id.btnA2toA1).setOnClickListener {
            finish()
        }

        findViewById<Button>(R.id.btnA2toA3).setOnClickListener {
            Intent(this, ThirdActivity::class.java).also {
                startActivity(it)
            }
        }
    }
}